src/main/java/hudson/plugins/git/util/BuildData.java

```diff
@@ -243,4 +243,22 @@ public String toString() {
                 ",buildsByBranchName="+buildsByBranchName+
                 ",lastBuild="+lastBuild+"]";
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (!(o instanceof BuildData)) {
+            return false;
+        } else {
+            BuildData otherBuildData = (BuildData) o;
+
+            if (otherBuildData.remoteUrls.equals(this.remoteUrls)
```

Comment:
@abayer, I'd much prefer to use the equals implementation that is null pointer safe and includes the hashCode implementation just in case some portion of the code is placing BuildData items into hash buckets.

Could you review https://github.com/MarkEWaite/git-plugin/commits/master-PR388-BuildData-equals and see if that is acceptable to you?